api/args/linkedin/profile/profile.go

```diff
@@ -0,0 +1,141 @@
+	ErrSeniorityNotSupported   = errors.New("seniority level not supported")
+	ErrFunctionNotSupported    = errors.New("function not supported")
+	ErrIndustryNotSupported    = errors.New("industry not supported")
+	ErrUnmarshalling           = errors.New("failed to unmarshal LinkedIn profile arguments")
+)
+
+const (
+	DefaultMaxItems    = 10
+	DefaultScraperMode = profile.ScraperModeShort
+	MaxItems           = 1000 // 2500 on the actor, but we will run over 1MB memory limit on responses
+)
+
+// Verify interface implementation
+var _ base.JobArgument = (*Arguments)(nil)
+
+// Arguments defines args for LinkedIn profile operations
+type Arguments struct {
+	Type                  types.Capability    `json:"type"`
+	ScraperMode           profile.ScraperMode `json:"profileScraperMode"`
+	Query                 string              `json:"searchQuery"`
+	MaxItems              uint                `json:"maxItems"`
+	Locations             []string            `json:"locations,omitempty"`
+	CurrentCompanies      []string            `json:"currentCompanies,omitempty"`
+	PastCompanies         []string            `json:"pastCompanies,omitempty"`
+	CurrentJobTitles      []string            `json:"currentJobTitles,omitempty"`
+	PastJobTitles         []string            `json:"pastJobTitles,omitempty"`
+	Schools               []string            `json:"schools,omitempty"`
+	YearsOfExperience     []experiences.Id    `json:"yearsOfExperienceIds,omitempty"`
+	YearsAtCurrentCompany []experiences.Id    `json:"yearsAtCurrentCompanyIds,omitempty"`
+	SeniorityLevels       []seniorities.Id    `json:"seniorityLevelIds,omitempty"`
+	Functions             []functions.Id      `json:"functionIds,omitempty"`
+	Industries            []industries.Id     `json:"industryIds,omitempty"`
+	FirstNames            []string            `json:"firstNames,omitempty"`
+	LastNames             []string            `json:"lastNames,omitempty"`
+	RecentlyChangedJobs   bool                `json:"recentlyChangedJobs,omitempty"`
+	StartPage             uint                `json:"startPage,omitempty"`
+}
+
+func (t *Arguments) UnmarshalJSON(data []byte) error {
+	type Alias Arguments
+	aux := &struct{ *Alias }{Alias: (*Alias)(t)}
+	if err := json.Unmarshal(data, aux); err != nil {
+		return fmt.Errorf("%w: %w", ErrUnmarshalling, err)
+	}
+	t.SetDefaultValues()
+	return t.Validate()
+}
+
+func (a *Arguments) SetDefaultValues() {
+	if a.MaxItems == 0 {
+		a.MaxItems = DefaultMaxItems
+	}
+	if a.ScraperMode == "" {
+		a.ScraperMode = DefaultScraperMode
+	}
+}
+
+func (a *Arguments) Validate() error {
+	var errs []error
+
+	if a.MaxItems > MaxItems {
+		errs = append(errs, ErrMaxItemsTooLarge)
+	}
+
+	err := a.ValidateCapability(types.LinkedInJob)
+	if err != nil {
+		errs = append(errs, err)
+	}
+
+	if !profile.AllScraperModes.Contains(a.ScraperMode) {
+		errs = append(errs, ErrScraperModeNotSupported)
+	}
+
+	for _, yoe := range a.YearsOfExperience {
+		if !experiences.All.Contains(yoe) {
```

Comment:
Re: all of these validations, should we return an error, or should we return the raw value? I'm thinking of the case where LinkedIn modifies their catalogs

Reply:
great call on future proofing, will add to the TODO comment about using a validation library (potentially)